lib/p2p/Peer.ts

```diff
@@ -234,17 +242,22 @@ class Peer extends EventEmitter {
    * Ensure we are connected (for inbound connections) or listen for the `connect` socket event (for outbound connections)
    * and set the [[connectTime]] timestamp. If an outbound connection attempt errors or times out, throw an error.
    */
-  private initConnection = async () => {
+  private initConnection = (retry?: boolean): Promise<void> => {
```

Comment:
Making this function `async` and not returning an explicit `Promise.resolve()` below was an intentional change that @michael1011 and I made in a separate PR. I think the code is a bit cleaner, that way. If you agree, line 252 would just `return` and 255 would return a `Promise<void>`.

Another option is to keep the method signature as is, and move the `if (this.connected)` logic into the `new Promise` logic, and then `resolve()` and `return` in there.

Reply:
I overrode this method signature in I resolved conflicts. Will return it to how it was on master.

Reply:
I overrode this method signature when I resolved conflicts. Will return it to how it was on master.